### ISSUE TITLE
Insert data with ajax with parametar

### ISSUE THREAD

**Issue by knubbe82** [yajra/laravel-datatables#1472]
### Summary of problem or feature request

I have company datatable with action button for editing, deleting. With third button I want to open page with company_id where I want to add clients. I am using ajax and modal form but i get MethodNotAllowedHttpException.

### Code snippet of problem

`Route::get('api/client/{id}', 'ClientController@apiClient')->name('api.client');`

```
public function apiClient($id)
    {
        $company = Company::find($id);

        $client = Client::all()->where('company_id', $id);

        return Datatables::of($client)
            ->addColumn('action', function($client)
            {
                return '<a href="#" class="btn btn-info btn-xs"><i class="glyphicon glyphicon-eye-open"></i></a> ' .
                    '<a onclick="editForm('. $client->id .')" class="btn btn-primary btn-xs"><i 
                    class="glyphicon glyphicon-edit"></i></a> ' .
                    '<a onclick="deleteData('. $client->id .')" class="btn btn-danger btn-xs"><i 
                    class="glyphicon glyphicon-trash"></i></a>';
            })->make(true);
    }

    public function clients($id)
    {
        $company = Company::find($id);

        return view('company.clientmaster')->with('kompanija', $company);
    }

public function store(Request $request)
    {
        $client = Client::create([
            'name'          =>  $request->name,
            'phone'         =>  $request->phone,
            'email'         =>  $request->email,
            'company_id'    =>  $request->company_id
        ]);

        return $client;

    }
```

> this is my script on view:

```
<script>
      var table = $('#subnettable').DataTable({
                  processing: true,
                  serverSide: true,
                  ajax: "{{ route('api.client', ['id' => $kompanija->id ]) }}",
                  columns: [
                    {data: 'name', name: 'name'},
                    {data: 'phone', name: 'phone'},
                    {data: 'email', name: 'email'},
                    {data: 'action', name: 'action', orderable: false, searchable: false}
                  ],
                  "language": {
                      "search": "Pretraži:",
                      "paginate": {
                          "first":      "Prva",
                          "last":       "Poslednja",
                          "next":       "Sledeća",
                          "previous":   "Predhodna"
                      },
                  "lengthMenu":     "Prikaži _MENU_",
                  "info":           "Prikazano je _START_ do _END_ od _TOTAL_ kompanija",
                  "infoFiltered":   "(pretraženo od _MAX_ ukupno kompanija)",
                  }
                });

      function addForm() {
        save_method = "add";
        $('input[name=_method').val('POST');
        $('#modal-form').modal('show');
        $('#modal-form form')[0].reset();
        $('.modal-title').text('Dodaj korisnika');
      }

$(function() {
        $('#modal-form form').validator().on('submit', function (e) {
          if (!e.isDefaultPrevented()) {
            var id = $('#id').val();
            if (save_method == 'add') url = "{{ url('client/store') }}";
            else url = "{{ url('client') . '/' }}" + id;

            $.ajax({
              url: url,
              type: "POST",
              data: $('#modal-form form').serialize(),
              success: function($data) {
                $('#modal-form').modal('hide');
                table.ajax.reload();
                swal({
                  title: 'Kompanija otvorena!',
                  text: 'Uspešno ste dodali kompaniju',
                  type: 'success',
                  timer: '1500'
              })
              },
              error: function() {
                swal({
                  title: 'Ufffff...',
                  text: 'Nešto nije u redu',
                  type: 'error',
                  timer: '1500'
              })
              }
            });
            return false;
          }
        });
      });
    </script>
```

> this is modal:

`<div class="modal" id="modal-form" tabindex="1" role="dialog" aria-hidden="true" data-backdrop="static">
    <div class="modal-dialog modal-lg" style="border: 3px solid #1f2d33">
        <div class="modal-content">
            <form method="post" class="form-horizontal" data-toggle="validator">
                {{ csrf_field() }} {{ method_field('POST') }}
                <div class="modal-header" style="background-color: #008bbf">
                    <button type="button" class="close" data-dismiss="modal" aria-label="Close">
                        <span aria-hidden="true"> &times; </span>
                    </button>
                    <h3 class="modal-title text-center" style="color: #fff"></h3>
                </div>

                <div class="modal-body" style="background-color: #ebf0f6">
                    <input type="hidden" id="id" name="id">
                    <div class="form-group">
                        <label for="name" class="col-md-3 control-label">Ime i prezime</label>
                        <div class="col-md-6">
                            <input type="text" id="name" name="name" class="form-control" autofocus required>
                            <span class="help-block with-errors"></span>
                        </div>
                    </div>

                    <div class="form-group">
                        <label for="phone" class="col-md-3 control-label">Telefon</label>
                        <div class="col-md-6">
                            <input type="text" id="phone" name="phone" class="form-control">
                            <span class="help-block with-errors"></span>
                        </div>
                    </div>
                   
                    <div class="form-group">
                        <label for="email" class="col-md-3 control-label">Email</label>
                        <div class="col-md-6">
                            <input type="email" id="email" name="email" class="form-control" required>
                            <span class="help-block with-errors"></span>
                        </div>
                    </div>
                    <input type="hidden" id="company_id" name="company_id" value="{{ $kompanija->id }}">
                </div>

                <div class="modal-footer">
                    <button type="submit" class="btn btn-primary btn-save">Snimi</button>
                    <button type="button" class="btn btn-default" data-dismiss="modal">Ponisti</button>
                </div>

            </form>
        </div>
    </div>
</div>`

### System details

- Operating System OS X El Capitan 10.11.6
- PHP Version 7.1.8
- Laravel Version 5.4
- Laravel-Datatables Version 8

**Comment by knubbe82** [comment]
I find solution :)
